core/include/io.h

```diff
@@ -40,6 +59,11 @@ static inline void io_write64(vaddr_t addr, uint64_t val)
 	*(volatile uint64_t *)addr = val;
 }
 
+#define io_write8_barrier(v, c)		({ __wmb(); io_write8(v, c); })
```

Comment:
I guess it might be a matter of taste, but in these situations, I'd prefer using static inline functions instead since they are specific with regard to types of arguments and return value. Same for the other function like macros below.